python/tvm/relay/op/contrib/vitis_ai.py

```diff
@@ -0,0 +1,92 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, unused-argument, no-else-return, E1102
+"""VITISAI codegen supported operators."""
+
+import numpy as np
+
+from tvm import relay
+import tvm._ffi
+from tvm.relay.expr import Tuple, TupleGetItem
+from tvm.relay import transform
+from tvm.relay.op.annotation import compiler_begin, compiler_end
+
+import pyxir
+import pyxir.frontend.tvm
+
+
+@transform.function_pass(opt_level=0)
+class VitisAIAnnotationPass:
```

Comment:
I'm interested as to why you use a custom annotation pass here. Is there something we could do to improve the standard annotation passes to make it work for your use case?

Reply:
@mbaret 
We could use the standard op based annotation, but our DPU supports one subgraph per model at this point of time.  If we use op based annotations for yolov3 kind of networks, it generates multiple subgraphs. To avoid multiple subgraph generation we would be using a custom annotation pass.

Reply:
Ah I see. Would I be right in understanding then that you could switch to the standard annotation mechanism once you have support for multiple subgraphs?

Reply:
Yes. Once we have support for multiple subgraphs we can use standard annotation pass.